pkg/state/statefile/statefile.go

```diff
@@ -100,7 +105,7 @@ func writeMetadataLen(w io.Writer, val uint64) error {
 // NewWriter returns a state data writer for a statefile.
 //
 // Note that the returned WriteCloser must be closed.
-func NewWriter(w io.Writer, key []byte, metadata map[string]string) (WriteCloser, error) {
+func NewWriter(w io.Writer, key []byte, metadata map[string]string, compress bool) (WriteCloser, error) {
```

Comment:
`compress` should be an enum, even if it currently only has two possible values.